### PR TITLE
Nano: fix tf bf16 conversion for functional API model with Conv2DTranspose layer

### PR DESCRIPTION
**Pull request by MeouSker77**
## Description

Fix tf bf16 conversion for functional API model with Conv2DTranspose layer. i.e. issue 1 in <https://github.com/analytics-zoo/nano/issues/318>

### 1. Why the change?

Model with Conv2DTranspose layer cannot be saved after modifying its dtype policy,
and model with custom object cannot be simply loaded by current code

### 2. User API changes

N/A

### 3. Summary of the change 

If save or load failed, use `from_config` to create bf16 model.

### 4. How to test?
- [ ] Unit test

### CODE REVIEW

**Comment by yangw1234** [comment]
Can we add a argument that allow user to pass their "custom object"?

**Comment by MeouSker77** [comment]
> Can we add a argument that allow user to pass their "custom object"?

Yes, will add this argument in next PR

**Review by Oscilloscope98 — APPROVED** [review]
LGTM

File: python/nano/src/bigdl/nano/tf/keras/amp/bfloat16.py [hunk 1]
```diff
@@ -13,25 +13,42 @@
...
+        # save or load operation may fail
         with TemporaryDirectory() as temp_dir:
             model.save(temp_dir)
+            # todo: fix loading custom object
             bf16_model = tf.keras.models.load_model(temp_dir)
+    except Exception as _e:
+        # if a functional model failed to save or load, try `from_config`
+        if hasattr(model, "input_shape"):
+            config = model.get_config()
+            bf16_model = tf.keras.Model.from_config(config)
```
Comment (rnwang04) [inline python/nano/src/bigdl/nano/tf/keras/amp/bfloat16.py:42]: why `tf.keras.Model.from_config(config)` will return a bf16 model?

**Review by rnwang04 — APPROVED** [review]
LGTM

**Pull request merged**